agent/stats/service_connect_linux.go

```diff
@@ -0,0 +1,90 @@
+//go:build linux
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package stats
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+
+	"github.com/cihub/seelog"
+
+	apitask "github.com/aws/amazon-ecs-agent/agent/api/task"
+)
+
+const (
+	statsPort         = ":9901/stats/prometheus"
+	bridgeNetworkMode = "bridge"
+)
+
+type ServiceConnectStats struct {
+	//TODO Change the type of Service Connect stats when it is defined.
+	stats string
+	lock  sync.RWMutex
+}
+
+func newServiceConnectStats() (*ServiceConnectStats, error) {
+	return &ServiceConnectStats{}, nil
+}
+
+func (sc *ServiceConnectStats) retrieveServiceConnectStats(task *apitask.Task, wg *sync.WaitGroup) {
+	defer wg.Done()
+	var ipAddress string
+	if task.IsNetworkModeAWSVPC() {
+		ipAddress = task.GetLocalIPAddress()
+	} else {
+		for _, container := range task.Containers {
+			// TODO: Implement a proper check for appnet agent container when it is defined
+			if strings.Contains(container.Name, "envoy") &&
+				(container.GetNetworkModeFromHostConfig() == "" || container.GetNetworkModeFromHostConfig() == bridgeNetworkMode) {
+				network := *container.GetNetworkSettings()
+				networkSettings := network.DefaultNetworkSettings
+				ipAddress = networkSettings.IPAddress
+				break
+			}
+		}
+	}
+	// TODO: Replace the stats url when it is defined
+	statsEndpoint, _ := url.Parse(fmt.Sprintf("http://" + ipAddress + statsPort))
```

Comment:
what if we don't find an ip address meaning we don't find the sc container?

Reply:
The method will print an error log "Could not connect to stats endpoint for task %s: %v", task.Arn, err" . Does that sound reasonable?